packages/sommelier.rb

```diff
@@ -2,84 +2,138 @@
-      system "echo 'GDK_BACKEND=x11' > .sommelier.env"
-      system "echo 'CLUTTER_BACKEND=wayland' >> .sommelier.env"
-      system "echo 'XDG_RUNTIME_DIR=/var/run/chrome' >> .sommelier.env"
-      system "echo 'WAYLAND_DISPLAY=wayland-0' >> .sommelier.env"
-      system "echo 'DISPLAY=:0' >> .sommelier.env"
-      system "echo 'SCALE=1' >> .sommelier.env"
-      system "echo '#!/bin/bash' > sommelierd"
-      system "echo 'sommelier -X --x-display=\$DISPLAY --scale=\$SCALE --no-exit-with-child /bin/sh -c \"#{CREW_PREFIX}/etc/sommelierrc\"' >> sommelierd"
-      system "echo '#!/bin/bash' > initsommelier"
-      system "echo 'SOMM=\$(pidof sommelier 2> /dev/null)' >> initsommelier"
-      system "echo 'if [ -z \"\$SOMM\" ]; then' >> initsommelier"
-      system "echo '  [ -f #{CREW_PREFIX}/bin/stopbroadway ] && stopbroadway' >> initsommelier"
-      system "echo '  #{CREW_PREFIX}/sbin/sommelierd &' >> initsommelier"
-      system "echo '  sleep 3' >> initsommelier"
-      system "echo 'fi' >> initsommelier"
-      system "echo 'SOMM=\$(pidof sommelier 2> /dev/null)' >> initsommelier"
-      system "echo 'if [ ! -z \"\$SOMM\" ]; then' >> initsommelier"
-      system "echo '  echo \"sommelier process \$SOMM is running\"' >> initsommelier"
-      system "echo 'else' >> initsommelier"
-      system "echo '  echo \"sommelier failed to start\"' >> initsommelier"
-      system "echo '  exit 1' >> initsommelier"
-      system "echo 'fi' >> initsommelier"
-      system "echo '#!/bin/bash' > stopsommelier"
-      system "echo 'SOMM=\$(pidof sommelier 2> /dev/null)' >> stopsommelier"
-      system "echo 'if [ ! -z \"\$SOMM\" ]; then' >> stopsommelier"
-      system "echo '  killall -g sommelier' >> stopsommelier"
-      system "echo '  if pgrep \"Xwayland\"; then killall Xwayland; fi'"
-      system "echo '  sleep 3' >> stopsommelier"
-      system "echo 'fi' >> stopsommelier"
-      system "echo 'SOMM=\$(pidof sommelier 2> /dev/null)' >> stopsommelier"
-      system "echo 'if [ -z \"\$SOMM\" ]; then' >> stopsommelier"
-      system "echo '  echo \"sommelier process stopped\"' >> stopsommelier"
-      system "echo 'else' >> stopsommelier"
-      system "echo '  echo \"sommelier process \$SOMM is running\"' >> stopsommelier"
-      system "echo '  exit 1' >> stopsommelier"
-      system "echo 'fi' >> stopsommelier"
+
+  system 'curl https://chromium.googlesource.com/chromiumos/platform2/+archive/be4e16feb380360cabbb5d6199a09592ecaf4a42.tar.gz | tar mzx --warning=no-timestamp'
+  Dir.chdir ("vm_tools/sommelier") do
+      # Using method here to download single file from googlesource in lieu of 
+      # checking out the ChromeOS kernel tree submodule:
+      #system 'curl "https://chromium.googlesource.com/chromiumos/third_party/kernel/+/refs/heads/master/include/uapi/linux/virtwl.h?format=TEXT" | base64 --decode > virtwl.h'
+
+
+    ######################### Download virtwl.h from Chromium 5.4 kernel tree ###########################################
```

Comment:
hmm... I'd have thought that my new linux headers package would've worked for this.

Reply:
This is a google modification used in the Chromium kernel fork which hasn't been submitted upstream yet, as far as I can tell.